core-common/src/main/java/org/zalando/nakadi/service/KPIEventMapper.java

```diff
@@ -45,27 +48,46 @@ public KPIEventMapper(final Set<Class<? extends KPIEvent>> mappedClasses) {
     }
 
     private Set<KPIGetter> findGettersOfKPIFields(final Class<? extends KPIEvent> mappedClass) {
+
+        // Take all methods that are public and has zero parameters, and map from method name to method.
+        final var getterCandidates = Arrays.stream(mappedClass.getMethods())
+                .filter(method -> method.getParameterCount() == 0)
+                .filter(method -> Modifier.isPublic(method.getModifiers()))
+                .collect(Collectors.toMap(Method::getName, Function.identity()));
+
         return Arrays.stream(mappedClass.getDeclaredFields())
                 .filter(field -> field.isAnnotationPresent(KPIEvent.KPIField.class))
-                .map(field -> findGetterByField(mappedClass, field))
+                .map(field -> mapFieldToKPIGetter(getterCandidates, field))
                 .collect(Collectors.toSet());
     }
 
-    private KPIGetter findGetterByField(final Class<? extends KPIEvent> clazz, final Field field) {
-        final var kpiField = field.getAnnotation(KPIEvent.KPIField.class);
-        final String getterName;
-        if (kpiField.getter().isEmpty()) {
+    private KPIGetter mapFieldToKPIGetter(final Map<String, Method> getterCandidates, final Field field) {
+        if (!SUPPORTED_FIELD_TYPES.contains(field.getType())) {
+            throw new InternalNakadiException("Type of " + field.getDeclaringClass().getName() + "." + field.getName()
+                    + " is not supported. Supported types are : " + SUPPORTED_FIELD_TYPES);
+        }
+        final var kpiFieldAnnotation = field.getAnnotation(KPIEvent.KPIField.class);
+        final Method getter;
+        if (kpiFieldAnnotation.getter().isEmpty()) {
             final var fieldName = field.getName();
-            getterName = "get" + fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
+            final var partialGetterName = fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
+            final var getName = "get" + partialGetterName;
+            final var isName = "is" + partialGetterName;
+            if (getterCandidates.containsKey(getName)) {
+                getter = getterCandidates.get(getName);
```

Comment:
Does it make sense to check for the `getterCandidate` method return type that matches with the field type?